plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -819,13 +820,19 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
             }
 
             external = true;
-            targetPath = getExternalLocationAsPath(externalLocation);
-            checkExternalPath(new HdfsContext(session), targetPath);
+            targetPath = Optional.of(getExternalLocationAsPath(externalLocation));
+            checkExternalPath(new HdfsContext(session), targetPath.get());
         }
         else {
             external = false;
-            LocationHandle locationHandle = locationService.forNewTable(metastore, session, schemaName, tableName, Optional.empty());
-            targetPath = locationService.getQueryWriteInfo(locationHandle).getTargetPath();
+            boolean isTransactional = isTransactional(tableMetadata.getProperties()).orElse(false);
```

Comment:
`isTransactional` should return `boolean`. i know, it's preexisting, so no change requested